### PR TITLE
Junit5 :: dir, id, & kff packages

### PR DESCRIPTION
**Pull request by dev-mlb**

### CODE REVIEW

File: src/test/java/emissary/directory/DirectoryEntryMapTest.java [hunk 1]
```diff
@@ -49,114 +49,113 @@ public void tearDown() throws Exception {
...
-            sb.append(i);
-            sb.append(".domain.com:7001/ThePlace");
-            this.dm.addEntry(new DirectoryEntry(sb.toString(), "blah", i % 100, 50));
+            String sb = forms[i % formsLength] +
+                    ".FOOPLACE." +
+                    phases[i % phasesLength] +
+                    ".http://host-" +
+                    i +
+                    ".domain.com:7001/ThePlace";
+            this.dm.addEntry(new DirectoryEntry(sb, "blah", i % 100, 50));
```
[the remark this answers is not in the document]
Comment (dev-mlb) [reply src/test/java/emissary/directory/DirectoryEntryMapTest.java:158]: I believe I had a sonarlint or ide recommendation to change it. Modern compilers end up using StringBuilder when doing string concatenation, so the two statements are essentially the same. If you run the test, the timing is very similar. The test is disabled anyways, so it really doesn't matter. 

I can change it back if you want?

File: src/test/java/emissary/directory/FTestPeerDirectoryPlace.java [hunk 2]
```diff
@@ -105,46 +108,46 @@ public void testDirectories() throws Exception {
...
-        assertNotNull("Directory entries in peer2", dl2);
-        assertEquals("Have entries for both peers", 2, dl1.size());
-        assertEquals("Have entries for both peers", 2, dl2.size());
+        assertNotNull(dl1, "Directory entries in peer1");
+        assertNotNull(dl2, "Directory entries in peer2");
+        assertEquals(2, dl1.size(), "Have entries for both peers");
+        assertEquals(2, dl2.size(), "Have entries for both peers");
 
         DirectoryEntry dl1d0 = dl1.getEntry(0);
         DirectoryEntry dl1d1 = dl1.getEntry(1);
```
[the remark this answers is not in the document]
Comment (dev-mlb) [reply src/test/java/emissary/directory/FTestPeerDirectoryPlace.java:139]: Yeah, the functional tests need an overhaul. We don't actually even run them - surefire runs all of the *Test.java files and failsafe will run all of the *IT.java files. We have nothing configured for functional tests, and most of them fail to run anyways. I started to clean up some of the functional tests, but we probably need a ticket to get them working again (or just remove them).

File: src/test/java/emissary/directory/DirectoryEntryMapTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (arp-0984) [reply src/test/java/emissary/directory/DirectoryEntryMapTest.java:158]: Thanks for your reply. It doesn't sound like a huge deal so its okay.

**Review by arp-0984 — APPROVED** [review]

File: src/test/java/emissary/directory/DirectoryEntryMapTest.java [hunk 3]
```diff
@@ -49,114 +49,113 @@ public void tearDown() throws Exception {
     }
 
     @Test
-    public void testAllEntries() {
+    void testAllEntries() {
         final List<DirectoryEntry> all = this.dm.allEntries();
-        assertNotNull("All entries not null", all);
-        assertEquals("All entries", 3, all.size());
+        assertNotNull(all, "All entries not null");
+        assertEquals(3, all.size(), "All entries");
```
Comment (andrewbp) [inline src/test/java/emissary/directory/DirectoryEntryMapTest.java:55]:
```suggestion
        assertEquals(3, all.size(), "Unexpected list size");
```

File: src/test/java/emissary/directory/DirectoryEntryMapTest.java [hunk 4]
```diff
@@ -49,114 +49,113 @@ public void tearDown() throws Exception {
     }
 
     @Test
-    public void testAllEntries() {
+    void testAllEntries() {
         final List<DirectoryEntry> all = this.dm.allEntries();
-        assertNotNull("All entries not null", all);
-        assertEquals("All entries", 3, all.size());
+        assertNotNull(all, "All entries not null");
```
Comment (andrewbp) [inline src/test/java/emissary/directory/DirectoryEntryMapTest.java:54]:
```suggestion
        assertNotNull(all, "List should not be null");
```

File: src/test/java/emissary/directory/KeyManipulatorTest.java [hunk 5]
```diff
@@ -1,168 +1,174 @@
...
         final String s = "UNKNOWN.FOOPLACE.ID.http://hosta.domain.com:8001/thePlace";
         final String wc = KeyManipulator.getDefaultDirectoryKey(s);
         final String hc = KeyManipulator.getHostMatchKey(s);
-        assertEquals("Default directory key", "*.*.*.http://hosta.domain.com:8001/DirectoryPlace", wc);
-        assertEquals("Host match key", "*.*.*.http://hosta.domain.com:8001/*", hc);
-        assertTrue("Defualt directory matches self", KeyManipulator.gmatch(s, hc));
-        assertTrue("Host match matches default directory", KeyManipulator.gmatch(wc, hc));
+        assertEquals("*.*.*.http://hosta.domain.com:8001/DirectoryPlace", wc, "Default directory key");
+        assertEquals("*.*.*.http://hosta.domain.com:8001/*", hc, "Host match key");
+        assertTrue(KeyManipulator.gmatch(s, hc), "Defualt directory matches self");
```
Comment (andrewbp) [inline src/test/java/emissary/directory/KeyManipulatorTest.java:123]:
```suggestion
        assertTrue(KeyManipulator.gmatch(s, hc), "Default directory matches self");
```

**Review by andrewbp — COMMENTED** [review]
There's a of messages that don't really describe why a test failed, but that feels like a separate task.

**Review by andrewbp — APPROVED** [review]

**Pull request merged**